nominatim/tools/special_phrases/sp_wiki_loader.py

```diff
@@ -0,0 +1,71 @@
+"""
+    Module containing the SPWikiLoader class.
+"""
+import re
+import logging
+from nominatim.tools.special_phrases.special_phrase import SpecialPhrase
+from nominatim.tools.special_phrases.sp_loader import SPLoader
+from nominatim.tools.exec_utils import get_url
+
+LOG = logging.getLogger()
+class SPWikiLoader(SPLoader):
+    """
+        Handles loading of special phrases from the wiki.
+    """
+    def __init__(self, config, languages=None):
+        if languages is not None and not isinstance(languages, list):
+            raise TypeError('The \'languages\' parameter should be of type list.')
+        super().__init__()
+        self.config = config
+        #Compile the regex here to increase performances.
+        self.occurence_pattern = re.compile(
+            r'\| *([^\|]+) *\|\| *([^\|]+) *\|\| *([^\|]+) *\|\| *([^\|]+) *\|\| *([\-YN])'
+        )
+        self.languages = self._load_languages() if not languages else languages
```

Comment:
Make a copy of the `languages` parameter here. You later manipulate the list and that might confuse the caller. (As a side effect, you can drop the check in line 16.)

Reply:
I am sorry I don't understand what you mean here, could you elaborate please?

Reply:
Lists are handed in by reference to a function. So, when I do something like:

```
langs = ['de' , 'en']
loader = SPWikiLoader(config, langs)
print(langs)
```
then, any modification you do to `languages` in your constructor, will be visible outside in the print, because they work on the same list. So when you consume `self.languages` with `pop()` below, the caller that called your constructor will see the list disappear as well.

This is not a problem in your particular code because the caller doesn't keep the list. It's still not good style because it is surprising and can lead to subtle problems in the future. The general rule is: if you keep a list you get handed into a function and/or modify it for your own private purposes, you should make a copy that does not interfere with what the caller later does with the list it has handed in. (Or make it very clear in the documentation that you modify/destroy the list, i.e. take ownership over it. But that would unnecessarily complicate things in this particular case.)

Reply:
Oh ok thanks, that make a lot more sense now. I was quite confuse at first as I didn't understand that you wanted me to copy the list.